datalad/core/distributed/push.py

```diff
@@ -114,8 +114,8 @@ class Push(Interface):
             constraints=EnsureStr() | EnsureNone(),
             doc="""specifies commit-ish (tag, shasum, etc.) from which to look for
             changes to decide whether pushing is necessary.
-            If an empty string is given, the last state of the current branch
-            at the sibling is taken as a starting point."""),
+            If '^' is given, the last state of the current branch at the sibling
+            is taken as a starting point."""),
```

Comment:
starting point for what? Unrelated to the intent of this PR, so I will just approve as is, but this option might need further explanation of "its purpose in life" - that unlike git " Using this option allows to avoid considering subdatasets where no changes were made since the that starting point" or alike

Reply:
https://github.com/datalad/datalad/issues/4626